ex3/ex3.c

```diff
@@ -2,14 +2,24 @@
 // the parent process should print "goodbye". You should ensure that the child
 // process always prints first.
 
+// > uses waitpid() to make sure the child process is executed first
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <sys/wait.h>
+#include <sys/wait.h> // this is needed for wait methods
 
 int main(void)
 {
     // Your code here
+    int forkreturns = fork(); // returns the pid of the child created by fork()
+
+    if (forkreturns == 0) {
```

Comment:
I see you've implemented fork() properly here. I'll assume the implementations in ex1 and 2 were just old attempts that haven't been changed yet :)